### PR TITLE
Rotary encoder with longpress and polling for mega328p

### PR DESCRIPTION
**Pull request by DeKi90**
I'm working on a Tonuino version that gets away with a single "button" - aka a KY-040 rotary encoder with push button. (And of course I am also using a classic Arduino nano for my project)
- Added "long press" detection to rotary encoder (more than 1 click)
- Added rotary encoder support for any pin on mega328p using timer interrupt and polling (hard coded at 200Hz)
- Added "long press" detection to rotary encoder
- Added platform.io config for classic_rotary

### CODE REVIEW

**Comment by boerge1** [comment]
Very nice idea to poll with a timer isr.
Will not merge your branch but will use some of your code.

**Comment by DeKi90** [comment]
Thanks @boerge1. I also saw that #159 might benefit from more generalized "soft" interrupts. Might be worth adding a function that mimics attachInterrupt to the project?

**Comment by boerge1** [comment]
@DeKi90 could you please describe what does the define LONGPRESS mean?
As I see, then commands are triggered only once per second and only if there are at least two clicks in this second.

File: src/constants.hpp [hunk 1]
```diff
@@ -42,10 +42,21 @@
...
+ * um den Drehgeber zu unterstützen bitte in der nächste Zeile den Kommentar entfernen
+ */
+// #define ROTARY_ENCODER
+
+/* uncomment the below line to enable the rotary encoder with long/short press like THREEBUTTON configuration
+ * um den Drehgeber mit lang/kurz Drehung wie 3-Knopf-Konfiguration zu benutzen bitte den Kommentar entferenen in der nächsten
+ * Zeile entfernen
  */
-//#define ROTARY_ENCODER
+// #define ROTARY_LONGPRESS
```
Comment (DeKi90) [inline src/constants.hpp:55]: @boerge1 With ROTARY_LONGPRESS the rotary encoder assumes both functions of the up/down buttons in the 3 button layout:
Short turn (1 click) -> up/down
Fast turn (2 or more clicks) -> upLong/downLong

**Comment by boerge1** [comment]
I implemented a version for common use with the above features on the branch issue_173. 
@DeKi90: Thanks for your ideas. 
This pull request can be closed.

**Pull request closed without merging**